### DISCUSSION TITLE
Ubuntu 20.04 Surface Laptop 4 AMD

### DISCUSSION THREAD

**Discussion by DrBrad** [linux-surface/linux-surface#581] [Q&A] [answered]
I was wanting to install Ubuntu 20.04 on my Microsoft Surface Laptop 4, however I noticed when testing Ubuntu my touchscreen, trackpad, and mouse dont work by default. I heard online that I need to change my kernel to yours, however I would need to install it first. Has anyone tested this or knows if I can still use my trackpad / keyboard when I change the kernel on Ubuntu.

**Comment by astejko** [comment, marked as answer]
Yea it works, just follow the [installation instructions](https://github.com/linux-surface/linux-surface/wiki/Installation-and-Setup), and check the [feature matrix](https://github.com/linux-surface/linux-surface/wiki/Supported-Devices-and-Features) to see what is supported. I have Ubuntu 21.04 on SL4A and I had to boot up with nomodeset, then followed the workarounds listed [here](https://github.com/linux-surface/linux-surface/wiki/Surface-Laptop-4) to get everything up and running properly.

**Reply by DrBrad** [reply]
https://github.com/linux-surface/linux-surface/wiki/Installation-and-Setup#surface-kernel-installation
What DKMS will I need to install and how do I do this, I'm sorry I have never installed Ubuntu on a laptop.

**Reply by qzed** [reply]
DKMS is a system for out-of-tree kernel modules. If you don't have any of those, you don't need to do anything. Common examples for such modules are ZFS and the nvidia driver.

If you have any of those installed, chances are that you'll notice that non-DKMS versions of the modules won't work. The kernel itself contains all the necessary drivers for the system to run, so you'd only be missing out on some extended functionality (e.g. the ZFS filesystem or GPU support for the nvidia card) if you don't have them installed. Also, you can always install them later anyways. The message here is just a heads-up that e.g. if you rely on ZFS as your root file system you may want to install the DKMS version or otherwise you won't be able to boot via the Surface kernel (you can always still choose the distro kernel from the bootloader and boot via that to fix things).

**Reply by DrBrad** [reply]
Okay, so will my built in GPU driver be installed with the kernel or is there a DKMS for this. It sounds like a DKMS is a driver for external / not out of the box products?

**Reply by qzed** [reply]
Yes that will work out of the box. All essential drivers you need are in the kernel.

DKMS is a system to distribute and build/install external kernel modules (the modules might contain drivers or other kernel extensions but DKMS itself is a driver/module). This is necessary, because kernel modules need to be built for the specific kernel version you intend to use them on. Due to that, distributions usually pre-build all external kernel modules they provide, which however means that they will only work with the kernel that the distribution provides. So if you install a custom kernel (e.g. the Surface kernel), you need a kernel module that is compiled for that specific custom kernel. DKMS essentially provides a way so that distributions can package the source/build artifacts of a kernel module and the DKMS system then builds and installs the module for the kernels you have locally when installing the package.

**Reply by DrBrad** [reply]
I tried installing Ubuntu 20.04 and it installed fine, however I am unable to boot to Ubuntu, I can still boot too windows because I setup Dual Boot, however I cant boot it to setup a kernel at this point.

Should I just take out the MVME and put it into another PC to setup the kernel or is there something I'm missing?

**Reply by qzed** [reply]
I believe you might need some kernel options to get the AMD version booting correctly. See 584 and https://github.com/linux-surface/linux-surface/wiki/Surface-Laptop-4#screen-freezes-with-corruption-on-boot. (You should be able to get it working without taking out the NVME.) Support for the AMD versions is unfortunately still sub-optimal (I believe that is not exclusive Surface devices but also affects a lot of newer AMD laptops).

**Reply by DrBrad** [reply]
Thank you @qzed it worked and I have Ubuntu running fine now. If anyone is unsure how to set "amd_iommu=off iommu=off" this is the process:

Boot to recovery

Select Root

Type  -  https://askubuntu.com/questions/19486/how-do-i-add-a-kernel-boot-parameter/19487#19487:
`sudo nano /etc/default/grub`

Change Line:
`GRUB_CMDLINE_LINUX_DEFAULT="quiet splash amd_iommu=off iommu=off  acpi_backlight=vendor"`

Type:
`sudo update-grub`

Save
Reboot regularly and follow the installation of linux-surface install.

Make sure to do:  `acpi_backlight=vendor`
Otherwise backlight changes wont work until you reboot.

**Reply by Sumitjh26997** [reply]
are there any updates on the AMD drivers - the workaround seems fine, but there are a lot of issues e.g. duplicate cursor and incredibly high GPU usage on video calls ?

**Reply by DrBrad** [reply]
@Sumitjh26997 
The duplicate cursor is an easy fix. It is an ubuntu issue not related to AMD actually haha.
Its the fractional scaling heres the fix.
```
sudo nano /etc/gdm3/custom.conf
uncomment out WaylandEnable=false (i.e. make it WaylandEnable=false)
sudo service gdm restart
```
The big issue is actually that suspend and hibernation are broken unless you have `nomodset`...
Otherwise I have seen no issues.

If hibernation and suspend are important to you then the deal is you have to disable the GPU otherwise accept that its broken until they fix it.

**Reply by DrBrad** [reply]
The WiFi has been very unstable sense 5.15, not sure if its my WiFi though haha.

**Comment by dels1975** [comment]
Hi,
I have a issue:
https://github.com/linux-surface/linux-surface/issues/1065

Who can help me?
